docs/design/auth.md

```diff
@@ -0,0 +1,60 @@
+# Runner Authentication and Authorization
+
+## Goals
+  - Support runner installs in untrusted domains.
+  - The account that configures or runs the runner process is not relevant for accessing GitHub resources.
+  - Accessing GitHub resources is done with a per-job token which expires when job completes.
+  - The token is granted to trusted parts of the system including the runner, actions and script steps specified by the workflow author as trusted. 
+
+## Configuration
+
+Configuring a self-hosted runner is [covered here in the documentation](https://help.github.com/en/actions/hosting-your-own-runners/adding-self-hosted-runners).
+
+Configuration is done with the user being authenticated via a time-limited, GitHub runner registration token.
+
+*Your credentials are never used for registering the runner with the service.*
+
+![Self-hosted runner config](../res/self-hosted-config.png)
+
+During configuration, an RSA public/private key pair is created, the private key is stored in file on disk. On Windows, the content is protected with DPAPI (machine level encrypted - runner only valid on that machine) and on Linux/OSX with chmod permissions.
+
+Using your credentials, the runner is registered with the service by sending the public key to the service which adds that runner to the pool and stores the public key, STS will generate clientId associated with the public key.
+
+## Start and Listen
+
+After configuring the runner, the runner can be started interactively (`./run.cmd` or `./run.sh`) or as a service.
+
+![Self-hosted runner start](../res/self-hosted-start.png)
+
+On start, the runner listener process loads the RSA private key (on windows decrypting with machine key DPAPI), sends a JWT token which signed by the private key to the service.
+The server response with an OAuth token that grants permission to access the message queue (HTTP long poll), allowing the runner to acquire the messages it will eventually run.
```

Comment:
Nit: responds

Reply:
It didn't highlight what I wanted to point out. "The server **responds** with an OAuth"

Reply:
Not sure if it's helpful, but this token is actually the same kind of token that is referenced as a "JWT Token" in the next section.